build-tools/osx/sign_and_notarize.sh

```diff
@@ -0,0 +1,231 @@
+#!/bin/bash
+
+set -e
```

Comment:
I'd also specify `set -o nounset`. Though it'll require refactoring the script a bit.